app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -230,6 +236,11 @@ class HomeFragment : Fragment() {
             ::dispatchModeChanges
         )
 
+        val blocklistProvider = LocalHomescreenBlocklistProvider(
+            lifecycleScope,
+            requireContext().homescreenBlocklistDataStore
+        )
```

Comment:
I think we tend to init component object lazily on the [Components](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/components/Components.kt)  class.

Reply:
Oh good call! I'll update this if the general design of the feature doesn't end up changing

Reply:
If it is supposed to be a global Singleton. Otherwise, if it's only used locally, then I think it makes sense to only create it in that scope.